pkg/serviceproxy/oidc_authenticator_test.go

```
@@ -0,0 +1,715 @@
+				User: &user.DefaultInfo{
+					Name:   "oidc:alice",
+					Groups: []string{"oidc:team-a"},
+				},
+			}, true, nil
+		}),
+		getImpersonateTokenFunc: func() (string, error) {
+			return "fake-sa-token", nil
+		},
+	}
+
+	req, _ := http.NewRequest("GET", "https://example.com/api", nil)
```

golangci-lint (2.12.2)

[error] net/http.NewRequest must not be called. use net/http.NewRequestWithContext

(noctx)